Description: Fix argv check
 Without this patch tint2 tries to access unallocated
 memory if the last argument is -c.
Author: Sebastian Reichel <sre@debian.org>
Bug-Debian: 716390
Last-Update: 2014-06-01
Applied-Upstream: yes

--- a/src/tint.c
+++ b/src/tint.c
@@ -127,16 +127,16 @@
 			printf("Usage: tint2 [-c] <config_file>\n");
 			exit(0);
 		}
-		if (!strcmp(argv[i], "-v") || !strcmp(argv[i], "--version"))	{
+		else if (!strcmp(argv[i], "-v") || !strcmp(argv[i], "--version"))	{
 			printf("tint2 version %s\n", VERSION_STRING);
 			exit(0);
 		}
-		if (!strcmp(argv[i], "-c")) {
+		else if (!strcmp(argv[i], "-c")) {
 			i++;
 			if (i < argc)
 				config_path = strdup(argv[i]);
 		}
-		if (!strcmp(argv[i], "-s"))	{
+		else if (!strcmp(argv[i], "-s"))	{
 			i++;
 			if (i < argc)
 				snapshot_path = strdup(argv[i]);
